### PR TITLE
folder_branch_ops: always send the initial MD straight to server

### PR DESCRIPTION
**Pull request by strib**
We don't want to return keys to chat that haven't yet been accepted by the server.

Issue: KBFS-2553

### CODE REVIEW

File: libkbfs/kbfs_ops.go [hunk 1]
```diff
@@ -439,6 +439,28 @@ func (fs *KBFSOpsStandard) GetTLFCryptKeys(
...
+		fb := FolderBranch{Tlf: rmd.TlfID(), Branch: MasterBranch}
+		ops := fs.getOpsNoAdd(ctx, fb)
+		lState := makeFBOLockState()
+		for ops.getLatestMergedRevision(lState) < kbfsmd.RevisionInitial {
+			err = ops.waitForJournal(ctx, lState, jServer)
+			if err != nil {
+				return nil, tlf.ID{}, err
+			}
+
+			rmd, err = fs.getMDByHandle(ctx, tlfHandle, FavoritesOpNoChange)
```
Comment (songgao) [inline libkbfs/kbfs_ops.go]: Is `rmd` here different from before the flush?

Comment (strib) [reply libkbfs/kbfs_ops.go]: Yes, it might now contain keys written by some other user who initialized the TLF simultaneously.

**Comment by strib** [comment]
Actually, I think this needs to be done inside `folderBranchOps.initMDLocked` instead, so that it applies to KBFS-related creates as well.  There's already a TODO there.  I'll fix it.

File: libkbfs/kbfs_ops.go (hunk 1 above)
Comment (songgao) [reply libkbfs/kbfs_ops.go]: Oh right. Thanks.

**Comment by strib** [comment]
Ok ready for review again, sorry for the churn.

**Comment by strib** [comment]
> doesn't the journal server also need to be told about the Put

Nope!  The journal server doesn't need to know about any merged revisions.  Works just the same as if this device started up after some other device had pushed the initial revision.

**Pull request merged**